src/number/numericString.ts

```diff
@@ -0,0 +1,32 @@
+export class NumericString {
+  radix: number;
+  value: number;
+  prefix: string;
+
+  private static matchings: { regex: RegExp, base: number, prefix: string }[] = [
+    { regex: /^0([0-7]+)$/, base: 8, prefix: "0"},
+    { regex: /^(\d+)$/, base: 10, prefix: ""},
+    { regex: /^0x([\da-fA-F]+)$/, base: 16, prefix: "0x"},
+  ];
+
+  static parse(input: string): NumericString | null {
+    for (let matcher of NumericString.matchings) {
```

Comment:
You can use destructuring here

`for (let { regex, base, prefix } of NumericString.matchings)`